pages/returning-members.js

```diff
@@ -0,0 +1,282 @@
+import FieldSection from "../frontend/components/molecules/Form/FieldSection";
+import Button from '../frontend/components/atoms/Button';
+
+const TERMDESCRIPTIONS = [
+"1A Co-op", 
+"1B Study", 
+"1B Co-op", 
+"2A Study", 
+"2A Co-op", 
+"2B Study", 
+'2B Co-op',
+"3A Study", 
+"3A Co-op",
+"3B Study", 
+"3B Co-op", 
+"4A Study",
+"4A Co-op", 
+"4B Study", 
+"4B Co-op", 
+"Graduate Student", 
+"Not currently enrolled at the University of Waterloo", 
+"Alumni",
+];
+
+const PREVTERMS = ['F22', 'S22', 'W22','F21', 'S21', 'W21', 'F20', "S20",'W20', 'F19', 'S19', 'W19', 'F18', 'S18', 'W18'];
+
+const FUTURETERMS = ['W22', 'S22', 'F22', 'W23', "S23",'F23', 'None of the above'];
+
+const SUBTEAMS = ["Business", "LIM", "Mechanical", "MC", "BMS", "Embedded","Infrastructure", "Web", "Team Hub"]
+
+const ACTIVESTATUS = [
+    "Yes, I will continue on the team, and I will be on campus (or working locally)", 
+    "Yes, I will continue on the team remotely and can come to campus if needed/possible", 
+    "Yes, I will continue on the team remotely only",
+    "Undecided or unsure", 
+    "No, taking the term off" 
+]
+
+const MEMBERSHIP = [
+    'Member', 
+    'Exec/lead/advisor', 
+    'Coop', 
+    'Not active on Waterloop this term',
+];
+
+const CONTINUINGSTATUS = [
+    'Continue with my sub-team', 
+    'Transfer to another sub-team (please specify)', 
+    'Want to take on a leadership role - lead', 
+    'Want to take on a leadership role - co-op supervisor', 
+    'Want to become a co-op', 
+    'I\'m undecided or not continuing',
+]
+
+const FormHeader = ({title, marginBottom}) => {
+    const theme = useContext(ThemeContext);
+    return (
+        <SystemComponent
+            fontSize={theme.fontSizes.header3}
+            textAlign='center'
+            mb={marginBottom}
+        >
+            <SystemSpan>
+                <Subtitle>{title}</Subtitle>
+            </SystemSpan>
+        </SystemComponent>
+    );
+}
+
+const RegistrationForm = () => {
+    const theme = useContext(ThemeContext);
+
+    const [formValues, setFormValues] = useState({
+        fullName:  "",
+        termDescription: "", 
+        previousTerms: [], 
+        futureTerms: [], 
+        subteam: "",
+        activeStatus: "", 
+        continuingStatus: "", 
+        email: "", 
+        comments: "",
+        futureTasks: "", 
+    });
+
+    const [hasError, setHasError] = useState({
+        fullName: false,
+        termDescription: false, 
+        membership: false, 
+        previousTerms: false, 
+        futureTerms: false,
+        activeStatus: false, 
+        subteam: false,
+        activeStatus: false,
+        continuingStatus: false,
+        email: false, 
+    });
+
+    const handleSave = () => {
+        console.log('clicked button')
+        console.log(formValues)
+        const updatedErrorList = {...hasError};
+
+        for (const [name, value] of Object.entries(updatedErrorList)) {
```

Comment:
Maybe you can add a // comment here saying why you set everything to false.

It's because we want to reset all fields to display no errors in the beginning.

If we don't do this, then the error text will remain on the fields, even if we corrected the errors for some of the fields. 

Make sense ?

Reply:
Either add an explanatory comment, or create a small function called `clearAllErrorMessages`. 

Either one of these measures will improve code readability here.